core/src/main/scala/org/http4s/Query.scala

```diff
@@ -197,14 +197,18 @@ object Query {
     *
     * If parsing fails, the empty [[Query]] is returned
     */
-  def fromString(query: String): Query =
+  def unsafeFromString(query: String): Query =
```

Comment:
If we're calling this unsafe, does it still make sense to quietly return an empty?  And might `fromString` now return an `Either`?

Reply:
I thought about doing that in two steps. first deprecating, then change the return type.
We could however do both in one step.

Reply:
I think we'll probably want to reclaim the real name `fromString` for 0.22 and 1.0, so we should probably just go ahead and do it in the next milestone.  But the change could be part of a subsequent PR.